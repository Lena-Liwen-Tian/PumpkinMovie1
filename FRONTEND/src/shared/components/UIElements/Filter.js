
import React,{useState} from 'react';
import Dropdown from 'react-bootstrap/Dropdown'
import DropdownButton from 'react-bootstrap/DropdownButton'
import Button from '../FormElements/Button';
const Filterbutton= props =>{

    const[Title,setTitle] = useState({"All":props.title});
    let endarr=[];
    Object.entries(props.elements).map(([key,value])=>{
    const changeValue = (key,value) =>{
      props.filterby(key,value);
      if(value !== "All") {
        setTitle({key:value});
      }else{
          setTitle({"All":props.title})
      }   
    }
    
    let arr=[];
    for (let i = 0; i <value.length; i++) {
        arr.push(<Dropdown.Item as="button" onClick={()=>changeValue(key,value[i])}>{value[i]}</Dropdown.Item>)
    }
    endarr.push(<DropdownButton id="dropdown-item-button" title={Title.key === undefined ?Title.All:Title.key}>
      {arr}
    </DropdownButton>)
   

 });
 
 
return (
 <table>
     <tr>
         {endarr.map(a=><td>{a}</td>)}
     </tr>
 </table>


)};
export default Filterbutton;