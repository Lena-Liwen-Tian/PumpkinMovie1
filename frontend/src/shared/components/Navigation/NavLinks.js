import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';

import { AuthContext } from '../../context/auth-context';
import './NavLinks.css';

const NavLinks = props => {
  const auth = useContext(AuthContext);
  let showtime = new Date();
  let m = showtime.getMonth()+1;
  let month = showtime.getMonth()+1>=10?"":"0";
  let showtimesId = showtime.getFullYear() + "-" + month + m +"-"+showtime.getDate(); 
  console.log(showtimesId)
  return (
    <ul className="nav-links">
      <li>
        <NavLink to="/movies" exact>
          Movies
        </NavLink>
      </li>
      <li>
          <NavLink to="/places">Theatres</NavLink>
      </li>     
         <li>
          <NavLink to={`/showtime/${showtimesId}`}>Show Times</NavLink>
        </li>
        <li>
        <NavLink to="/reviews" exact>Reviews</NavLink>
      </li>
     
      <li>
        <NavLink to="/" exact>
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/aboutus" exact>
          About Us
        </NavLink>
      </li>
      <li>
        <NavLink to="/cart" exact>
          Cart
        </NavLink>
      </li>

      
    </ul>
  );
};

export default NavLinks;
