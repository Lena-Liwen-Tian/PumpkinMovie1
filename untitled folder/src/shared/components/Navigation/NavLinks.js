import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';

import { AuthContext } from '../../context/auth-context';
import './NavLinks.css';

const NavLinks = props => {
  const auth = useContext(AuthContext);
 
  return (
    <ul className="nav-links">
      <li>
        <NavLink to="/movies" exact>
          Movies
        </NavLink>
      </li>
      <li>
          <NavLink to="/places">Theatres</NavLink>
      </li>     
         <li>
          <NavLink to={`/showtime/2020-03-22`}>Show Times</NavLink>
        </li>
  
     
      <li>
        <NavLink to="/" exact>
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/aboutus" exact>
          About Us
        </NavLink>
      </li>

      {!auth.isLoggedIn && (
        <li>
          <NavLink to="/auth">Authenticate</NavLink>
        </li>
      )}
      {auth.isLoggedIn && (
        <li>
          <button onClick={auth.logout}>Logout</button>
        </li>
      )}

      
    </ul>
  );
};

export default NavLinks;
