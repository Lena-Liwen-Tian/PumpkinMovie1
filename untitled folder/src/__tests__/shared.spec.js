import Button from '../shared/components/FormElements/Button';
import React from 'react';
import Card from '../shared/components/UIElements/Card';
import { create } from "react-test-renderer";
import { Link } from 'react-router';
import { mount, render,shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainHeader from '../shared/components/Navigation/MainHeader';
import NavLinks from '../shared/components/Navigation/NavLinks';
import Avatar from '../shared/components/UIElements/Avatar';
import Filter from '../shared/components/UIElements/Filter';
import PaginationPage from '../shared/components/UIElements/pagination';
import SearchButton from '../shared/components/UIElements/Search';
import Sortbutton from '../shared/components/UIElements/Sort';
configure({ adapter: new Adapter() });

describe("test button",()=>{
    it("Matches the snapshot",()=>{
        const button = create(<Button />);
        expect(button.toJSON()).toMatchSnapshot();
    })
    it("renders the button with link to",()=>{
        const container1 =shallow(<Button to="/">Click</Button>);       
        const link = container1.find("Link");       
        expect(link.props().to).toEqual("/");          
    })
    it("renders the button with href",()=>{
    const container2 = shallow(<Button href="/">Click</Button>);
    const a = container2.find("a");
    expect(a.props().href).toEqual("/");
    });
    it("renders regular button",()=>{
        const container3 = shallow(<Button>Click</Button>);
        const b = container3.find("button");
        expect(b.props().children).toEqual("Click");
    });

})
    describe("test MainHeader",()=>{
    it("render header correctly",()=>{
        const header = shallow(<MainHeader>children</MainHeader>);
        expect(header.find("header").length).toBe(1);
    })
    })
    describe("test NavLinks",()=>{
        it("render NavLinks correctly",()=>{
            const links = shallow(<NavLinks/>);
            expect(links.find("ul").length).toBe(1);
        })
    

})
describe("test Avatar",()=>{
    it("render Avatar correctly",()=>{
        const avatar = shallow(<Avatar className="no" style="no" image="no" alt="no" width="no" width="no"/>);
        expect(avatar.find("img").length).toBe(1);
    })
})

describe("test Card",()=>{
    it("render Card correctly",()=>{
        let card = shallow(<Card className="no" style="no">children</Card>);
        expect(card.find("div").length).toBe(1);
    })
})
describe("test Filter",()=>{
    it("render filter correctly",()=>{
        let e = {"time":"no"};
        let filter = shallow(<Filter elements ={e} title="no"/>);
        expect(filter.find("tr").length).toBe(1);
    })
    
})
describe("test pagination",()=>{
    it("render pagination correctly",()=>{
        let page = shallow(<PaginationPage totalPost="2" postsPerPage="2" currntpage ="1" paginate="1" />)
        expect(page.find("span").length).toBe(1);
    })
    
})
describe("test search",()=>{
    it("render search correctly",()=>{
        let search = shallow(<SearchButton placeholder="no" />)
        expect(search.find("input").length).toBe(1);
    })
    
})
describe("test sort",()=>{
    it("render search correctly",()=>{
        let item=["no"];
        let sort = shallow(<Sortbutton items={item} sortby="no" />)
        expect(sort.find("DropdownButton").length).toBe(1);
    })
    
})

