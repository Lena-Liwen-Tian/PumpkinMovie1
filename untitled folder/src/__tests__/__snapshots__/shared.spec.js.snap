// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button Matches the snapshot 1`] = `
<button
  className="button button--default undefined undefined"
/>
`;
